### PR TITLE
Add TDB_OPT_CONS_NO_BIGRAMS to disable bigram optimization.

### PR DESCRIPTION
**Pull request by Noeda**
@tuulos 

This can significantly decrease the time it takes to finalize a TrailDB at tdb_cons_finalize() at the cost of increase in TrailDB filesize, with no change in the actual contents of the TrailDB.

### CODE REVIEW

File: src/tdb_cons.c [hunk 1]
```diff
@@ -728,6 +728,9 @@ TDB_EXPORT tdb_error tdb_cons_set_opt(tdb_cons *cons,
                 default:
                     return TDB_ERR_INVALID_OPTION_VALUE;
             }
+        case TDB_OPT_CONS_NO_BIGRAMS:
+            cons->no_bigrams = !(!(value.value));
```
Comment (tuulos) [inline src/tdb_cons.c]: why the double-negation?

File: src/tdb_encode_model.c [hunk 2]
```diff
@@ -411,11 +412,13 @@ tdb_error make_grams(FILE *grouped,
     TDB_TIMER_END("encode_model/find_candidates")
 
     /* collect frequencies of *all* occurring bigrams of candidate unigrams */
-    TDB_TIMER_START
-    ret = event_fold(all_bigrams, grouped, num_events, items, num_fields, &g);
-    if (ret)
-        goto done;
-    TDB_TIMER_END("encode_model/all_bigrams")
+    if (!no_bigrams) {
```
Comment (tuulos) [inline src/tdb_encode_model.c]: The `choose_grams` function below is an `O(n^2)` operation, which should be the most expensive part here. If there are no bigrams, the `choose_grams` could be made much less expensive.

File: src/tdb_cons.c (hunk 1 above)
Comment (Noeda) [inline src/tdb_cons.c]: To force the value to be 0 or 1

File: src/tdb_encode_model.c (hunk 2 above)
Comment (Noeda) [inline src/tdb_encode_model.c]: I don't remember if I specifically checked between these two functions but I do know that this particular change I made has quite big impact.

I wasn't sure if `choose_grams` is something that can be skipped. Maybe I will try removing both/or trying to remove `choose_grams` only.

File: src/tdb_cons.c (hunk 1 above)
Comment (tuulos) [inline src/tdb_cons.c]: for readability, you could change this to `(value.value ? 1: 0)`.

File: src/tdb_encode_model.c (hunk 2 above)
Comment (tuulos) [inline src/tdb_encode_model.c]: ok, if this PR makes a big impact as such, no reason to delay it further. We can add the further optimization in `choose_grams` later. Could you please add a comment in the `!no_bigrams` block ```
/* TODO: choose_grams below could be optimized for the nobigram case */
```

**Comment by tuulos** [comment]
@Noeda two more things for this PR:
- Take a look at https://github.com/traildb/traildb/blob/master/tests/c-tests/tdb_test.h - `test_cons_settings()` gets executed in all tests using `cons` and it allows us to run all tests over different parametrization of `cons`. Please add an env var for `NO_BIGRAMS` and make sure it gets applies to all tests, and they pass.
- Update docs: CHANGELOG and `tdb_cons_set_opt` about the new option

**Comment by Noeda** [comment]
I've made the changes but didn't have to go through tests yet so don't merge anything yet.

**Pull request merged**